<a href="https://colab.research.google.com/github/papaymaguire/ece219-project2/blob/main/project2/notebooks/part01_01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import userdata
gh_pat = userdata.get('gh_pat')
gh_username = userdata.get('gh_username')

In [2]:
!rm -rf ece219-project2
!git clone https://{gh_username}:{gh_pat}@github.com/papaymaguire/ece219-project2.git

Cloning into 'ece219-project2'...
remote: Enumerating objects: 40, done.
remote: Counting objects: 100% (40/40), done.
remote: Compressing objects: 100% (29/29), done.
remote: Total 40 (delta 7), reused 21 (delta 3), pack-reused 0
Receiving objects: 100% (40/40), 13.23 KiB | 1.65 MiB/s, done.
Resolving deltas: 100% (7/7), done.


In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
import sys
sys.path.insert(0,'/content/ece219-project2')

import pandas as pd
import numpy as np

In [5]:
from project2.utils.GoogleDriveDataIO import GoogleDriveDataIO
drive_io = GoogleDriveDataIO("/content/drive", "My Drive/EC ENGR 219/Project 2/Data")

In [6]:
# Silences warnings from sklearn
def warn(*args, **kwargs):
    pass
import warnings
warnings.warn = warn

# Question 1

In [7]:
from sklearn.datasets import fetch_20newsgroups
from sklearn.feature_extraction.text import TfidfVectorizer
class_1_cats = ["comp.graphics", "comp.os.ms-windows.misc", "comp.sys.ibm.pc.hardware", "comp.sys.mac.hardware"]
class_2_cats = ["rec.autos", "rec.motorcycles", "rec.sport.baseball", "rec.sport.hockey"]
newsgroups_train = fetch_20newsgroups(subset='all', remove=('headers', 'footers'), categories=class_1_cats + class_2_cats)
vectorizer = TfidfVectorizer(min_df=3, stop_words="english")
X_tfidf = vectorizer.fit_transform(newsgroups_train.data)
X_tfidf.shape

(7882, 23522)

The resulting TF-IDF matrix has 7882 rows and 23522 columns corresponding to 7882 samples and 23522 features extracted from those samples.

In [8]:
newsgroups_train.target.shape

(7882,)